viewing_party/main.py

```diff
@@ -0,0 +1,292 @@
+    for movie in watched_list:
+        genre = movie["genre"]
+        # if the genre does not exist in the genre_counts dict, initialize it to 0
+        #genre_count keys is the same a s
+        if (genre not in genre_count.keys()):
+            #populate the genre_count dict: add this key value pair 
+            genre_count[genre] = 0
+        
+        # for every one genre_count that gets populated add 1 to reflect the number of times it appears in this dictionary
+        genre_count[genre] += 1
+        
+    # find the nax value of the genre_counts.values() ==> max frequency
+    maximum_genre_counts = max(genre_count.values())
+
+    # Find the genre name that corresponds to the max frequency count
+    for genre in genre_count:
+        if genre_count[genre] == maximum_genre_counts:
+                 
+    # return genre_name
+            return genre       
+
+    return None
+
+#----------------------------WAVE 3----------------------------------------------------------------------
+
+def get_unique_watched(user_data):
+    #create a copy of the users watched data as i will remove items from it and dont want it to affect the 
+    #original data
+    users_watched_movies_copy = user_data["watched"].copy()
+    
+    #iterate over the user_data's friends dictionary
+    for friend_dict in user_data["friends"]:
+        
+        #create a new variable to store the friends watched dictionary
+        friend_movies_list = friend_dict["watched"]
+        
+        #iterate over the friends watched movie list to get the elements which is in dict format
+        for friend_movie in friend_movies_list:
+            
+            #if the movie exists in the list call the helper funtions with the given parameters is true then:
+            if (does_movie_exist_in_list(friend_movie["title"], users_watched_movies_copy)):
+                
+                #then remove the movie by calling the helper function
+                remove_duplicate_movie_from_list(friend_movie["title"], users_watched_movies_copy)
+    
+    return users_watched_movies_copy         
+
+
+def get_friends_unique_watched(user_data):
+    #initailize an empty list to store the output
+    friends_unique_watched = []
+    
+    #create a new variable to store the users watched dictionaries
+    users_watched_movies = user_data["watched"]
+    
+    #iterate over the users friends data to ge the friends dictionary 
+    for friend_dict in user_data["friends"]:
+        
+        #create a new variabe to store the friends watched list of dictionaries
+        friend_movies_list = friend_dict["watched"]
+        
+        #iterate over the friends watched list of dictionary (each element is a dictionary)
+        for friend_movie in friend_movies_list:
+            
+            #call the helper function to check if the statement is true: if the movie does not exist is false
+            if (not does_movie_exist_in_list(friend_movie["title"], users_watched_movies)):
```

Comment:
This is a clever use of a helper function. Something to consider is how readable is this? You might also consider something like `if friend_movie["title'] not in users_watched_movies`